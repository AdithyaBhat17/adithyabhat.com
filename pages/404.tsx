import Container from '@/components/container'
import Head from '@/components/header'
import { motion } from 'framer-motion'
import animationData from 'lotties/Dog_news_paper.json'
import Link from 'next/link'
import Lottie from 'react-lottie'

const defaultOptions = {
  loop: true,
  animationData,
  rendererSettings: {
    preserveAspectRatio: 'xMidYMid slice',
  },
}

function PageNotFound() {
  return (
    <Container>
      <Head title="404 - Page not found" />
      <Lottie options={defaultOptions} width={300} height={300} />
      <motion.div
        initial={{ opacity: 0, y: 10 }}
        animate={{ opacity: 1, y: 0 }}
        layout
        className="mx-auto text-left lg:text-center leading-relaxed space-y-2"
      >
        <h1 className="text-4xl leading-normal poppins mb-4 font-semibold">
          Oops, Page not found!
        </h1>
        <p className="text-lg poppins">Marv just tore this page apart ðŸ¥º</p>
        <br />
        <Link legacyBehavior href="/">
          <a className="button_primary poppins mt-5 text-lg">Go Home</a>
        </Link>
      </motion.div>
    </Container>
  )
}

export default PageNotFound
