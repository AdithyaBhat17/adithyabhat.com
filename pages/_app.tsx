import Footer from '@/components/footer'
import Navbar from '@/components/navbar'
import * as gtag from '@/lib/gtag'
import { SpeedInsights } from '@vercel/speed-insights/next'
import { Router } from 'next/router'
import NProgress from 'nprogress'
import '../styles/index.css'

const handleRouteChange = (url: string) => {
  gtag.pageview(url)
}

Router.events.on('routeChangeStart', (url: string) => {
  NProgress.start()
  handleRouteChange(url)
})
Router.events.on('routeChangeComplete', (url: string) => {
  NProgress.done()
  handleRouteChange(url)
})
Router.events.on('routeChangeError', (url: string) => {
  NProgress.done()
  handleRouteChange(url)
})

function MyApp({ Component, pageProps }) {
  return (
    <div className="flex flex-col justify-around min-h-screen">
      <Navbar />
      <main className="p-0">
        <Component {...pageProps} />
      </main>
      <div className="mt-24 lg:mt-32">
        <Footer />
      </div>
      <SpeedInsights />
    </div>
  )
}

export default MyApp

export function reportWebVitals({ id, name, label, value }) {
  ;(window as any).gtag('event', name, {
    event_category: label === 'web-vital' ? 'Web Vitals' : 'Next.js metric',
    value: Math.round(name === 'CLS' ? value * 1000 : value), // values must be integers
    event_label: id, // id unique to current page load
    non_interaction: true, // avoids affecting bounce rate.
  })
}
