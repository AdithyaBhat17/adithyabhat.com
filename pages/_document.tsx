/* istanbul ignore file */

import { GA_TRACKING_ID } from '@/lib/gtag'
import Document, {
  Html,
  Main,
  NextScript,
  Head,
  DocumentContext,
} from 'next/document'

export default class MyDocument extends Document {
  static async getInitialProps(ctx: DocumentContext) {
    const initialProps = await Document.getInitialProps(ctx)
    return { ...initialProps }
  }
  render() {
    return (
      <Html lang="en">
        <Head>
          <link
            rel="preconnect"
            href="https://fonts.gstatic.com/"
            crossOrigin="anonymous"
          />
          <link
            rel="preconnect"
            href="https://cdn.dribbble.com/"
            crossOrigin="anonymous"
          />
          <link
            rel="preconnect"
            href="https://vitals.vercel-analytics.com"
            crossOrigin="anonymous"
          />
          <link
            rel="preconnect"
            href="https://www.datocms-assets.com"
            crossOrigin="anonymous"
          />

          <script
            async
            src={`https://www.googletagmanager.com/gtag/js?id=${GA_TRACKING_ID}`}
          />
          <script
            dangerouslySetInnerHTML={{
              __html: `
            window.dataLayer = window.dataLayer || [];
            function gtag(){dataLayer.push(arguments);}
            gtag('js', new Date());
            gtag('config', '${GA_TRACKING_ID}', {
              page_path: window.location.pathname,
            });
          `,
            }}
          />
          <script
            type="application/ld+json"
            dangerouslySetInnerHTML={{
              __html: `{
                "name":"Adithya NR",
                "description":"I am a Full stack Designer based in Bengaluru, India.",
                "author":{
                  "@type":"Person",
                  "name":"Adithya NR"
                },
                "disambiguatingDescription": "Web Designer & Developer",
                "jobTitle": "Full Stack Designer, Freelancer",
                "@type":"WebSite",
                "url":"https://adithyabhat.com/",
                "publisher":{
                  "@type":"Organization",
                  "logo":{
                    "@type":"ImageObject",
                    "url":"https://raw.githubusercontent.com/AdithyaBhat17/adithyabhat17.github.io/version-3/public/adithya1.png"
                  },
                    "name":"Adithya NR"
                },
                "image":"https://raw.githubusercontent.com/AdithyaBhat17/adithyabhat17.github.io/version-3/public/adithya1.png",
                "headline":"Adithya NR",
                "sameAs":[
                  "https://twitter.com/adithya__nr",
                  "https://linkedin.com/in/adithya-nr/",
                  "https://github.com/AdithyaBhat17",
                  "https://instagram.com/adithyabhat__/",
                  "https://medium.com/adithya-nr"
                ],
                "alternateName": "Adithya Bhat",
                "@context":"http://schema.org"
              }`,
            }}
          />
        </Head>
        <body>
          <Main />
          <NextScript />
        </body>
        <script type="text/javascript" src="/viewport.js"></script>
      </Html>
    )
  }
}
