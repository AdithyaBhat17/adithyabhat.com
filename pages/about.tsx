import Head from '@/components/header'
import Container from '@/components/container'
import Link from 'next/link'
import { companies } from '@/utils/about'
import { motion } from 'framer-motion'
import { fadeInUp } from '@/utils/motion'
import useResetScroll from 'hooks/useResetScroll'
import Image from 'next/image'
import { calculateAge } from '@/utils/age'

export default function About({ companies, age }) {
  useResetScroll()
  return (
    <motion.div initial="initial" animate="animate" exit={{ opacity: 0 }}>
      <Head title="About - Adithya NR" />
      <Container className="2xl:px-96">
        <motion.div variants={fadeInUp} className="px-0 md:px-16 xl:px-32">
          <div className="flex justify-center">
            <Image
              width={250}
              height={250}
              loading="eager"
              className="block mx-auto"
              src="/static/about_hero.svg"
              alt="About Adithya"
            />
          </div>
          <h1 className="text-3xl md:text-4xl my-10 text-left sm:text-center font-semibold">
            About me
          </h1>
          <p className=" my-5 leading-8">
            As you might have guessed already, I&apos;m Adithya, a{' '}
            <span data-testid="age">{age}</span>
            -year-old UX Engineer from Bengaluru, India.
          </p>
          <p className="my-5 leading-8">
            I&apos;m a React nanodegree graduate and an IDF certified UX
            Designer. Free time is hard to come by, but I freelance and craft
            web apps for a living when it does.
          </p>
          <h2 className="text-md leading-8 uppercase opacity-50 mt-5 mb-3 font-semibold">
            Background
          </h2>
          <p className="my-5 leading-8">
            I studied computer science and engineering at RNS Institute of
            Technology, Bangalore. During my time at RNSIT, I volunteered as a
            graphic designer with a small team that helped me jump-start and
            hone my digital designer skills.
          </p>
          <p className="my-5 leading-8">
            I&apos;m currently working on the{' '}
            <a
              href="https://www.softway.com/b2e"
              target="_blank"
              rel="noopener noreferrer"
              className="left text-blue-800 font-medium"
            >
              Culture+â„¢
            </a>{' '}
            product suite at{' '}
            <a
              href="https://www.softway.com/"
              target="_blank"
              rel="noopener noreferrer"
              className="left text-blue-800 font-medium"
            >
              Softway
            </a>{' '}
            as a UX Engineer. Besides design, I also build web apps using React,
            Node, and many more tools.
          </p>
          <h3 className="text-xl md:text-3xl my-20 text-left sm:text-center font-semibold">
            Companies I&apos;ve worked with
          </h3>
          <div className="flex flex-wrap mx-auto justify-between sm:justify-around items-center mb-5">
            {companies?.map((company) => (
              <div key={company.name} className="w-1/2 md:w-1/5 mx-auto my-5">
                <img
                  loading="lazy"
                  className="mx-auto mb-10 w-1/2 sm:w-2/3 "
                  src={company.path}
                  alt={company.name}
                  title={company.name}
                />
              </div>
            ))}
          </div>
          <h3 className="text-xl md:text-3xl mt-20 mb-16 text-left sm:text-center font-semibold">
            Things I&apos;m good at
          </h3>
          <img
            loading="lazy"
            className="mx-auto w-3/4 md:w-1/3 block h-56 sm:h-auto"
            src="/static/skills.svg"
            alt="Things I'm good at"
          />
          <div className="flex flex-wrap mx-auto text-left md:text-center mt-16 md:mt-24">
            <div className="w-full mx-auto sm:flex-1">
              <h2 className="inline-flex items-center text-xl font-semibold">
                <img
                  src="/static/pen-tool.svg"
                  className="mr-5"
                  alt="Pen tool icon"
                />{' '}
                Design
              </h2>{' '}
              <br />
              <small className="block text-md sm:w-4/5 sm:mx-auto leading-7 my-5">
                User Experience, User Interface, Wireframing, Prototyping, Web
                design
              </small>
            </div>
            <div className="w-full mx-auto flex-1">
              <h2 className="inline-flex items-center text-xl font-semibold">
                <img
                  src="/static/terminal.svg"
                  className="mr-5"
                  alt="Terminal icon"
                />{' '}
                Development
              </h2>{' '}
              <br />
              <small className="block text-md sm:w-4/5 sm:mx-auto leading-7 my-5">
                JavaScript, Reactjs, Redux, Express, Nodejs, TypeScript, Nextjs,
                GraphQL, PostgreSQL
              </small>
            </div>
          </div>
          <h1 className="text-2xl md:text-5xl font-semibold my-20 md:my-40 leading-relaxed md:leading-relaxed">
            Got a project in mind, a question or something else? Get in touch
            with me{' '}
            <Link href="/contact">
              <a className="left"> here</a>
            </Link>
          </h1>
        </motion.div>
      </Container>
    </motion.div>
  )
}

export function getStaticProps() {
  const age = calculateAge()
  return {
    props: {
      companies,
      age,
    },
  }
}
