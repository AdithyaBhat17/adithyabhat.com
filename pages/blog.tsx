import { fetchAPI } from '@/lib/datocms'
import { ALL_ARTICLES_QUERY } from '../graphql/queries/posts'

import Head from '@/components/header'
import Navbar from '@/components/navbar'
import Container from '@/components/container'
import { BlogPosts } from '@/interfaces/blog'
import ArticlesList from '@/components/articles-list'
import Footer from '@/components/footer'
import { motion } from 'framer-motion'

export default function Blog({ data }: BlogPosts) {
  return (
    <motion.div initial="initial" exit={{ opacity: 0 }} animate="animate">
      <Head title="Blog | Adithya NR" />
      <Navbar />
      <Container>
        <motion.h1
          initial={{ opacity: 0, y: 60 }}
          animate={{ opacity: 1, y: 0 }}
          className="font-semibold text-5xl text-gray-900 mb-6"
        >
          <b>Blogs</b>
        </motion.h1>
        <ArticlesList data={data} />
      </Container>
      <br />
      <Footer />
    </motion.div>
  )
}

// istanbul ignore next line
export async function getStaticProps() {
  const allArticles = await fetchAPI(ALL_ARTICLES_QUERY)
  return {
    props: { data: allArticles },
  }
}
