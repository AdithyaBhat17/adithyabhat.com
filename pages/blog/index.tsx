import { fetchAPI } from '@/lib/datocms'
import { ALL_ARTICLES_QUERY } from '../../graphql/queries/posts'

import Head from '@/components/header'
import Container from '@/components/container'
import { List } from '@/interfaces/content'
import CardsList from '@/components/cards'
import { motion } from 'framer-motion'

export default function Blog({ data, type, columns }: List) {
  return (
    <motion.div initial="initial" exit={{ opacity: 0 }} animate="animate">
      <Head title="Blog | Adithya NR" />
      <Container>
        <motion.h1
          initial={{ opacity: 0, y: 60 }}
          animate={{ opacity: 1, y: 0 }}
          className="font-semibold text-5xl text-gray-900 mb-6"
        >
          Articles
        </motion.h1>
        <CardsList data={data} type={type} columns={columns} />
      </Container>
    </motion.div>
  )
}

// istanbul ignore next line
export async function getStaticProps() {
  const allArticles = await fetchAPI(ALL_ARTICLES_QUERY)
  return {
    props: { data: allArticles, type: 'allArticles', columns: '3' },
  }
}
