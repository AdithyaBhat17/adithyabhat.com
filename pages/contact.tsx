import Navbar from '@/components/navbar'
import Head from '@/components/header'
import Container from '@/components/container'
import { useMemo, useState, BaseSyntheticEvent, useEffect } from 'react'
import Footer from '@/components/footer'

import { useForm } from 'react-hook-form'
import FormErrorMessage from '@/components/error-message'
import { ContactData, ContactResponse } from '@/interfaces/contact'
import { motion } from 'framer-motion'
import { fadeInUp, stagger } from '@/utils/motion'

export default function Contact() {
  const { register, handleSubmit, errors } = useForm()
  const [status, setStatus] = useState('idle')

  useEffect(() => {
    if (status === 'success' || status === 'error') {
      setTimeout(() => setStatus('idle'), 3000)
    }
  }, [status])

  const submitMessage = async (
    data: ContactData,
    event: BaseSyntheticEvent
  ) => {
    const { name, email, message } = data
    if (!name || !email || !message.trim()) {
      return
    }
    try {
      setStatus('sending')
      const contactResponse = await fetch(`/api/contact`, {
        method: 'POST',
        body: JSON.stringify({
          name: name.trim(),
          email,
          message: message.trim(),
        }),
      })
      const contactStatus: ContactResponse = await contactResponse.json()
      if (contactStatus.success) {
        setStatus('success')
        event.target.reset()
      } else {
        setStatus('error')
      }
    } catch (error) {
      setStatus('error')
    }
  }

  const buttonText = useMemo(() => {
    if (status === 'success') return 'Message sent!'
    else if (status === 'error') return 'Mission Failed 😶'
    else if (status === 'sending') return 'Sending message'
    return 'Send message'
  }, [status])

  return (
    <motion.div initial="initial" animate="animate" exit={{ opacity: 0 }}>
      <Head title="Contact - Adithya NR" />
      <Navbar />
      <Container>
        <div className="md:w-2/3 w-full mx-auto">
          <motion.h1
            initial={{ opacity: 0 }}
            animate={{ opacity: 1 }}
            className="md:text-5xl text-3xl font-bold mb-3"
          >
            Get in touch
          </motion.h1>
          <motion.p initial={{ opacity: 0 }} animate={{ opacity: 1 }}>
            Send me a message here and I&apos;ll get back to you ASAP
          </motion.p>
          <motion.form
            variants={stagger}
            onSubmit={handleSubmit(submitMessage)}
          >
            <motion.div variants={fadeInUp} className="form-group">
              <label className="label" htmlFor="name">
                Your name
              </label>
              <input
                className="input"
                type="text"
                ref={register({ required: 'Name cannot be empty 🙁' })}
                name="name"
                id="name"
                placeholder="Mike Wazowski"
              />
              {errors.name ? (
                <FormErrorMessage message={errors.name.message} />
              ) : null}
            </motion.div>
            <motion.div variants={fadeInUp} className="form-group">
              <label className="label" htmlFor="email">
                Your email address
              </label>
              <input
                className="input"
                type="email"
                ref={register({
                  required: 'Please provide your email address 😓',
                  pattern: {
                    value: /[^@ \t\r\n]+@[^@ \t\r\n]+\.[^@ \t\r\n]+/,
                    message: 'Invalid email address...',
                  },
                })}
                name="email"
                id="email"
                placeholder="mike@monstersinc.com"
              />
              {errors.email ? (
                <FormErrorMessage message={errors.email.message} />
              ) : null}
            </motion.div>
            <motion.div variants={fadeInUp} className="form-group">
              <label className="label" htmlFor="message">
                Your message
              </label>
              <textarea
                ref={register({
                  required: 'Please leave a message 😢',
                  minLength: { value: 10, message: 'Message too short 😞' },
                })}
                className="input"
                name="message"
                id="message"
                rows={5}
              />
              {errors.message ? (
                <FormErrorMessage message={errors.message.message} />
              ) : null}
            </motion.div>
            <motion.input
              variants={fadeInUp}
              whileHover={{ scale: 1.05, x: 0 }}
              whileTap={{ scale: 0.5, x: 0 }}
              type="submit"
              role="submit"
              value={buttonText}
              className={`submit-button w-full sm:w-auto ${
                status === 'success' ? 'bg-green-600 text-black' : ''
              }`}
            />
          </motion.form>
        </div>
      </Container>
      <br />
      <Footer />
    </motion.div>
  )
}
