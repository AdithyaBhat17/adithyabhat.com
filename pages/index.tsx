import ArticlesList from '@/components/cards'
import Head from '@/components/header'
import Hero from '@/components/hero'
import Work from '@/components/work'
import { AllArticles, AllProjects, List } from '@/interfaces/content'
import { fetchAPI } from '@/lib/datocms'
import Container from '@/components/container'
import NextLink from '@/components/NextLink'
import { motion } from 'framer-motion'
import useResetScroll from 'hooks/useResetScroll'
import Thanks from '@/components/thanks'
import { HOME_PAGE_QUERY } from 'graphql/queries'
import useResizeObserver from 'hooks/useResizeObserver'
import { useEffect } from 'react'

// eslint-disable-next-line no-undef
export const Home = ({ data }: List): JSX.Element => {
  useResetScroll()
  const { width } = useResizeObserver()

  useEffect(() => {
    if (window !== undefined && width) {
      const isTablet = width >= 768 && width <= 1024

      const limit = isTablet ? 4 : 3

      ;(data as AllArticles).allArticles = (data as AllArticles).allArticles?.slice(
        0,
        limit
      )
    }
  }, [width])

  return (
    <motion.div initial="initial" animate="animate" exit={{ opacity: 0 }}>
      <Head title="Adithya NR | A Full stack Designer based in Bengaluru, India." />
      <Hero />
      <Work data={data as AllProjects} />
      <Container>
        <h1 className="mt-24 xl:mt-64 mb-6 text-2xl font-semibold">
          Featured articles
        </h1>
        <ArticlesList
          data={data as AllArticles}
          type="allArticles"
          columns="3"
        />
        <NextLink href="/blog" text="Read more articles" />
        <Thanks />
      </Container>
    </motion.div>
  )
}

export async function getStaticProps() {
  const data: List = await fetchAPI(HOME_PAGE_QUERY)
  return {
    props: {
      data,
    },
  }
}

export default Home
