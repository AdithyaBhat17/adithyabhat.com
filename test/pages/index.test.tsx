import React from 'react'
import { render } from '../testUtils'
import { Home } from '../../pages/index'
import { cleanup } from '@testing-library/react'
import MyApp from '../../pages/_app'
import { mockArticles } from '../__mocks__/article'
import About from '@/pages/about'

afterEach(cleanup)

describe('Home page', () => {
  it('matches snapshot', () => {
    const data = mockArticles
    const { asFragment } = render(
      <Home recentWork="" featuredArticles={data} />
    )
    expect(asFragment()).toMatchSnapshot()
  })
  it('Imports css modules using _app.tsx', () => {
    render(<MyApp Component={About} pageProps={{}}></MyApp>)
  })
})
