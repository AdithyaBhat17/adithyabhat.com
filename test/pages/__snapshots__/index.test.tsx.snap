// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="jsx-2250570153 container"
  >
    <main
      class="jsx-2250570153"
    >
      <h1
        class="jsx-2250570153 title"
      >
        Welcome to 
        <a
          class="jsx-2250570153"
          href="https://nextjs.org"
        >
          Next.js!
        </a>
      </h1>
      <p
        class="jsx-2250570153 description"
      >
        Get started by editing 
        <code
          class="jsx-2250570153"
        >
          pages/index.tsx
        </code>
      </p>
      <button
        class="jsx-2250570153"
      >
        Test Button
      </button>
      <div
        class="jsx-2250570153 grid"
      >
        <a
          class="jsx-2250570153 card"
          href="https://nextjs.org/docs"
        >
          <h3
            class="jsx-2250570153"
          >
            Documentation →
          </h3>
          <p
            class="jsx-2250570153"
          >
            Find in-depth information about Next.js features and API.
          </p>
        </a>
        <a
          class="jsx-2250570153 card"
          href="https://nextjs.org/learn"
        >
          <h3
            class="jsx-2250570153"
          >
            Learn →
          </h3>
          <p
            class="jsx-2250570153"
          >
            Learn about Next.js in an interactive course with quizzes!
          </p>
        </a>
        <a
          class="jsx-2250570153 card"
          href="https://github.com/vercel/next.js/tree/master/examples"
        >
          <h3
            class="jsx-2250570153"
          >
            Examples →
          </h3>
          <p
            class="jsx-2250570153"
          >
            Discover and deploy boilerplate example Next.js projects.
          </p>
        </a>
        <a
          class="jsx-2250570153 card"
          href="https://vercel.com/import?filter=next.js&utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
        >
          <h3
            class="jsx-2250570153"
          >
            Deploy →
          </h3>
          <p
            class="jsx-2250570153"
          >
            Instantly deploy your Next.js site to a public URL with Vercel.
          </p>
        </a>
      </div>
    </main>
    <footer
      class="jsx-2250570153"
    >
      <a
        class="jsx-2250570153"
        href="https://vercel.com?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
        rel="noopener noreferrer"
        target="_blank"
      >
        Powered by 
        <img
          alt="Vercel Logo"
          class="jsx-2250570153 logo"
          src="/vercel.svg"
        />
      </a>
    </footer>
  </div>
</DocumentFragment>
`;
